### ISSUE TITLE
nodecore blocks- N102: DOES_NOT_FIT warning - startup failure

### ISSUE THREAD

**Issue by Hou5e** [VeriBlock/nodecore-releases#46]
It looks like nodecore-0.2.1-r2 tried to skip a block (15536)???  Loading blocks with warning N102: DOES_NOT_FIT after that, and eventually gives up and stops processing the blockchain. It didn't seem like it re-tried immediately when the error happened, like the warning message said it would, and tried to continue on:

Log messages:
2018-07-18 19:05:58,382 INFO [main] n.core.a.k [DefaultBlockchain.java:1367] Before applying block 15535(00000000138AD6B68AB687E5D7F51876DF41B82E3EFC5F90), latest Bitcoin block known: 00000000D8386D19CF08FEB3A5B1A28C7FB3C9945A201E26DFAB001180AA9369
2018-07-18 19:05:58,382 INFO [main] n.core.a.k [DefaultBlockchain.java:1393] When adding block 15535 (00000000138AD6B68AB687E5D7F51876DF41B82E3EFC5F90), we are also adding 4 Bitcoin blocks.
2018-07-18 19:05:58,397 INFO [main] n.core.a.k [DefaultBlockchain.java:1508] Added a new block (00000000138AD6B68AB687E5D7F51876DF41B82E3EFC5F90 @ 15535) to best chain.
2018-07-18 19:05:58,397 INFO [main] n.core.a.k [DefaultBlockchain.java:1517] Calculating block template for block 15536!
2018-07-18 19:05:58,397 INFO [main] n.core.a.k [DefaultBlockchain.java:181] Took 0ms to add block #15535
2018-07-18 19:05:58,397 INFO [main] n.core.a.k [DefaultBlockchain.java:1367] Before applying block 15537(0000000003661C7D26D5D413A719E84A85D4101703A62B02), latest Bitcoin block known: 00000000D8386D19CF08FEB3A5B1A28C7FB3C9945A201E26DFAB001180AA9369
2018-07-18 19:05:58,427 WARN [main] n.core.a.k [DefaultBlockchain.java:96] **N102: DOES_NOT_FIT**
2018-07-18 19:05:58,427 WARN [main] n.core.a.k [DefaultBlockchain.java:97]     Block has been placed in the cache and another attempt will be made to re-add it
2018-07-18 19:05:58,427 INFO [main] n.core.a.k [DefaultBlockchain.java:181] Took 0ms to add block #15537
2018-07-18 19:05:58,427 INFO [main] n.core.a.k [DefaultBlockchain.java:1367] Before applying block 15538(0000000002880B761F9B8E82219722DD598327093FF89C84), latest Bitcoin block known: 00000000D8386D19CF08FEB3A5B1A28C7FB3C9945A201E26DFAB001180AA9369
2018-07-18 19:05:58,442 WARN [main] n.core.a.k [DefaultBlockchain.java:96] **N102: DOES_NOT_FIT**
.
.
.

--
Additional log file details:
[nodecore-block-DOES_NOT_FIT-error.txt](https://github.com/VeriBlock/nodecore-releases/files/2208080/nodecore-block-DOES_NOT_FIT-error.txt)

**Comment by Hou5e** [comment]
Restarting nodecore again: It stops at block 15535, and says it's done, and doesn't proceed:

20:51:49.529 [main] INFO  nodecore.core.a.k - Calculating block template for block 15535!
20:51:49.559 [main] INFO  nodecore.core.a.k - Took 0ms to add block #15534
20:51:49.559 [main] INFO  nodecore.core.a.k - Before applying block 15535(00000000138AD6B6
8AB687E5D7F51876DF41B82E3EFC5F90), latest Bitcoin block known: 00000000D8386D19CF08FEB3A5B
1A28C7FB3C9945A201E26DFAB001180AA9369
20:51:49.559 [main] INFO  nodecore.core.a.k - When adding block 15535 (00000000138AD6B68AB
687E5D7F51876DF41B82E3EFC5F90), we are also adding 4 Bitcoin blocks.
20:51:49.574 [main] INFO  nodecore.core.a.k - Added a new block (00000000138AD6B68AB687E5D
7F51876DF41B82E3EFC5F90 @ 15535) to best chain.
20:51:49.579 [main] INFO  nodecore.core.a.k - Calculating block template for block 15536!
20:51:49.579 [main] INFO  nodecore.core.a.k - Took 0ms to add block #15535
**20:51:49.599 [main] INFO  nodecore.core.a.k - Blockchain save started
20:51:49.649 [main] INFO  nodecore.core.a.k - Blockchain save completed
20:51:49.654 [main] INFO  nodecore.g - NodeCore blockchain state is now: Loaded
20:51:49.834 [main] INFO  nodecore.g.s - Saving 1 addresses to wallet transaction cache
20:51:49.969 [main] INFO  nodecore.g.s - Saving 1386 transaction to wallet transaction cac
he
20:51:50.199 [main] INFO  nodecore.f.c - Starting Admin RPC service using security mode "N
one" on /127.0.0.1:10501 with password disabled
20:51:50.499 [main] INFO  nodecore.f.d - Starting Peer service on /0.0.0.0:6501
20:51:50.504 [main] INFO  nodecore.d.m - Initializing peer table
20:51:50.504 [main] INFO  nodecore.g - NodeCore operating state is now: Running**
20:52:09.744 [pool-1-thread-1] INFO  nodecore.Program - ** Minute Marker: 16 **
20:52:09.744 [pool-1-thread-1] INFO  nodecore.Program - Free Memory: 364946KB | MaxMemory:
 2796544KB | TotalMemory: 2321920KB
20:52:10.534 [pool-4-thread-1] INFO  nodecore.d.i - Opening connection to 45.32.207.123
20:52:10.549 [pool-4-thread-1] INFO  nodecore.d.i - Opening connection to 207.148.122.103
20:52:10.554 [pool-4-thread-1] INFO  nodecore.d.i - Opening connection to 144.202.49.232
20:52:10.554 [pool-4-thread-1] INFO  nodecore.d.i - Opening connection to 207.148.85.146
20:53:09.744 [pool-1-thread-1] INFO  nodecore.Program - ** Minute Marker: 17 **
20:53:20.634 [pool-1-thread-1] INFO  nodecore.Program - Free Memory: 327433KB | MaxMemory:
 2796544KB | TotalMemory: 2321920KB
20:54:09.750 [pool-1-thread-1] INFO  nodecore.Program - ** Minute Marker: 18 **
20:54:09.750 [pool-1-thread-1] INFO  nodecore.Program - Free Memory: 327420KB | MaxMemory:
 2796544KB | TotalMemory: 2321920KB
20:55:09.750 [pool-1-thread-1] INFO  nodecore.Program - ** Minute Marker: 19 **
20:55:09.750 [pool-1-thread-1] INFO  nodecore.Program - Free Memory: 327420KB | MaxMemory:
 2796544KB | TotalMemory: 2321920KB
20:56:09.750 [pool-1-thread-1] INFO  nodecore.Program - ** Minute Marker: 20 **
20:56:09.750 [pool-1-thread-1] INFO  nodecore.Program - Free Memory: 327406KB | MaxMemory:
 2796544KB | TotalMemory: 2321920KB

**Comment by JoshRack** [comment]
We found a scenario in which some blocks were not getting saved to disk, and the result is a gap, then the blocks that appear after the gap all trip the "DOES NOT FIT" validation. Normally, the node would sync with the network and fill in the gap, but as you indicated, the bootstrapping nodes haven't been reliably accepting new connections.

We believe we've fixed the saving to disk problem in the next release of NodeCore and we've got some things in the works for the bootstrapping nodes as well.
